internal/translation/privateendpoint/privateendpoint.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2024 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package privateendpoint
+
+import (
+	"context"
+	"strings"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
+)
+
+const (
+	ErrorServiceNotFound = "PRIVATE_ENDPOINT_SERVICE_NOT_FOUND"
+)
+
+type PrivateEndpointService interface {
+	ListPrivateEndpoints(ctx context.Context, projectID, provider string) ([]EndpointService, error)
+	GetPrivateEndpoint(ctx context.Context, projectID, provider, ID string) (EndpointService, error)
+	CreatePrivateEndpointService(ctx context.Context, projectID string, peService EndpointService) (EndpointService, error)
+	DeleteEndpointService(ctx context.Context, projectID, provider, ID string) error
+	CreatePrivateEndpointInterface(ctx context.Context, projectID, provider, serviceID, gcpProjectID string, peInterface EndpointInterface) (EndpointInterface, error)
+	DeleteEndpointInterface(ctx context.Context, projectID, provider, serviceID, ID string) error
+}
```

Comment:
nit pick

These seem to be 2 separate interfaces that are used together, one for `EndpointService` types and another for `EndpointInterfaces`

I would rather separate both, even if at the high level are still joined together. It brings more clarity:
```go
type PrivateEndpointService interface {
  PrivateEndpoints
  PrivateInterfaces
}
```
Naming is tricky, not suggesting this naming scheme suggestion is the best BTW.

Reply:
I actually started with this approach, but because of the confusion generated by possible names and their meaning (each provider has one), I decided that a unified interface would be clearer than a separate one.